### PR TITLE
Connector caching

### PR DESCRIPTION
**Pull request by asvetlov**
rename TCPConnector DNS caching API

### CODE REVIEW

File: aiohttp/connector.py [hunk 1]
```diff
@@ -419,10 +422,21 @@ def __init__(self, *, verify_ssl=True, fingerprint=None,
             self._hashfunc = hashfunc
         self._fingerprint = fingerprint
 
+        if cache_dns is not _marker and resolve is not _marker:
+            if cache_dns != resolve:
+                raise ValueError("cashe_dns must agree with resolve")
```
Comment (kxepal) [inline aiohttp/connector.py]: s/cashe/cache/

File: aiohttp/connector.py [hunk 2]
```diff
@@ -467,30 +481,57 @@ def family(self):
         return self._family
 
     @property
+    def cache_dns(self):
```
Comment (kxepal) [inline aiohttp/connector.py]: `cache_dns` assumes some action, here is "to cache dns". But for property either `dns_cache` or `cached_dns` are better names.

Comment (kxepal) [inline aiohttp/connector.py]: Especially since below you use "cached_hosts" and "clear_dns_cache" terminology.

File: aiohttp/connector.py [hunk 3]
```diff
@@ -467,30 +481,57 @@ def family(self):
...
+            raise ValueError("either both host and port "
+                             "or none of them are allowed")
+        else:
+            self._cached_hosts.clear()
+
+    @property
     def resolve(self):
         """Do DNS lookup for host name?"""
-        return self._resolve
+        warnings.warn((".resolve property is deprecated, "
```
Comment (kxepal) [inline aiohttp/connector.py:511]: If resolve is deprecated, may be raise warning in `__init__` as well?

Comment (asvetlov) [inline aiohttp/connector.py:511]: Done

File: aiohttp/connector.py (hunk 1 above)
Comment (asvetlov) [inline aiohttp/connector.py]: Done

File: aiohttp/connector.py (hunk 2 above)
Comment (asvetlov) [inline aiohttp/connector.py]: Renamed `cache_dns` to `dns_cache`.
Do you suggest renaming ctor parameter also?

Comment (kxepal) [inline aiohttp/connector.py]: Worth to. Btw, on second thought I found myself incorrect about the name: seems like this property is used as True/False flag in order to define will dns cache be used or not, right? Then `use_dns_cache` should be semantically better and won't cause confusion about if it some sort of cache mapping or not. What do you think?

Comment (asvetlov) [inline aiohttp/connector.py]: use_dns_cache is the best, sure.
Thanks for suggestion

On Sat, Jun 20, 2015 at 4:37 PM, Alexander Shorin notifications@github.com
wrote:

> In aiohttp/connector.py
> https://github.com/KeepSafe/aiohttp/pull/415#discussion_r32884576:
> 
> > @@ -467,30 +481,57 @@ def family(self):
> >          return self._family
> > 
> > ```
> >  @property
> > ```
> > -    def cache_dns(self):
> 
> Worth to. Btw, on second thought I found myself incorrect about the name:
> seems like this property is used as True/False flag in order to define will
> dns cache be used or not, right? Then use_dns_cache should be
> semantically better and won't cause confusion about if it some sort of
> cache mapping or not. What do you think?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/KeepSafe/aiohttp/pull/415/files#r32884576.

## 

Thanks,
Andrew Svetlov

**Comment by asvetlov** [comment]
I guess now everything is done.

**Pull request merged**